app/models/concerns/hyku_addons/account_behavior.rb

```diff
@@ -106,6 +109,25 @@ def public_settings
       settings.reject { |k, _v| Account.private_settings.include?(k.to_s) }
     end
 
+    def shared_search_enabled?
+      Flipflop.enabled?(:cross_tenant_shared_search)
+    end
+
+    def shared_search_tenant?
+      ActiveModel::Type::Boolean.new.cast(shared_search)
+    end
+
+    def add_parent_id_to_child
+      create_child_parent_association_from_submitted_tenant_ids
+      settings['tenant_list'] = []
+      save
+    end
+
+    def remove_existing_child_records
+      return self unless children.present?
```

Comment:
I think you can change the method for the following:

```suggestion
      self.children = []
      #Not sure if this is necessary
      save
```